activemq-broker/src/main/java/org/apache/activemq/broker/scheduler/SchedulerBroker.java

```diff
@@ -418,6 +426,12 @@ public void scheduledJob(String id, ByteSequence job) {
             producerExchange.setMutable(true);
             producerExchange.setProducerState(new ProducerState(new ProducerInfo()));
             super.send(producerExchange, messageSend);
+
+            if(jobSchedulerActivityListener != null) {
+                Message msg = messageSend.copy();
+                msg.setProperty(ScheduledMessage.AMQ_SCHEDULER_ACTIVITY, ScheduledMessage.AMQ_SCHEDULER_ACTIVITY_DISPATCHED);
+                jobSchedulerActivityListener.dispatched(this, msg);
+            }
```

Comment:
@mattrpav Once message is dispatched it has scheduledJobId property, so I can use that, but I figured it is nice to have a single source to monitor scheduler activity.